sklearn/metrics/cluster/tests/test_unsupervised.py

```diff
@@ -169,19 +169,22 @@ def test_non_numpy_labels():
 
 
 def test_silhouette_nonzero_diag():
+    # Make sure silhouette_samples requires diagonal to be zero.
+    # Non-regression test for #12178
+
     # Construct a zero-diagonal matrix
     dists = pairwise_distances(
```

Comment:
Would it make sense to parametrize for both 32 and 64 bits floating point precision?

Reply:
sure  why not, done